docs/contracts/transient-storage.rst

```diff
@@ -0,0 +1,166 @@
+.. index:: ! transient storage, ! transient, tstore, tload
+
+.. _transient-storage:
+
+*****************
+Transient Storage
+*****************
+
+Transient storage is another data location besides memory, storage, calldata
+(and return-data and code) which was introduced alongside its respective opcodes
+``TSTORE`` and ``TLOAD`` by `EIP-1153 <https://eips.ethereum.org/EIPS/eip-1153>`_.
+This new data location behaves as a key-value store similar to storage with the main
+difference being that data in transient storage is not permanent, but is scoped to
+the current transaction only, after which it will be reset to zero.
+The values in transient storage are never deserialized from storage or serialized to storage,
+thus the gas cost of operating in transient storage is much cheaper,
+since it doesn't require disk access.
+EVM version ``cancun`` or newer is required for transient storage to be available.
+
+Transient storage variables cannot be initialized since the value would be cleared at
+the end of the transaction, rendering the initialization ineffective.
```

Comment:
```suggestion
the end of the creation transaction, rendering the initialization ineffective.
```

There would, of course, be the option to initialize at the beginning of each call frame before the dispatch - but that may be unexpected, resp. expensive, so we also opt not to do it for now. Not sure we should elaborate on it here, though.